crates/core_simd/src/vector.rs

```diff
@@ -311,6 +311,42 @@ where
         unsafe { self.store(slice.as_mut_ptr().cast()) }
     }
 
+    #[must_use]
+    #[inline]
+    pub fn masked_load_or(slice: &[T], or: Self) -> Self {
+        Self::masked_load_select(slice, Mask::splat(true), or)
+    }
+
+    #[must_use]
+    #[inline]
+    pub fn masked_load_select(slice: &[T], enable: Mask<isize, N>, or: Self) -> Self {
+        let ptr = slice.as_ptr();
+        let idxs = Simd::<usize, N>::from_slice(&[
```

Comment:
I don't like this. The `rotate_lanes_left` code does a clever thing with a constexpr and maybe that should be preferred?

But ideally, I'd like to have something like `Simd<T, N>::lane_indices() -> Simd<T, N>` (generating 0 through N elements) built-in as it's a useful function to have in user code.